### ISSUE TITLE
Consider checking/building all func bodies

### ISSUE THREAD

**Issue by dominikh** [dominikh/go-tools#37]
Currently, we only check and build the function bodies of directly checked packages, not of imported dependencies. This makes the tool run faster. In the past, we didn't ever need to look into function bodies of dependencies. Nowadays, however, we compute purity (and soon always-nil errors) of functions, and that information is useful for dependencies, too.

Benchmark how much slower checking all bodies is.

**Comment by mvdan** [comment]
I looked into the same for `interfacer` - although it never needs to check past func signatures in non-checked packages.

A couple of thoughts:

* If the package has a large dependency tree, type-checking time can easily go up 4x or more (e.g. if it imports `net/http`)
* If you want to gather info from those non-checked func bodies, perhaps you could do this for all of `std` in a `go generate`, meaning that only non-checked non-std packages would need to be done on top of the current ones

**Comment by dominikh** [comment]
I've made the change locally and saw about a 60% increase in runtime for checking all of cockroachdb, which is acceptable, and can most likely be counteracted with some performance improvements elsewhere. 

I don't think that overall this will be a big problem, for two reasons: 1) the standard library itself checks rather quickly (~7 seconds here) 2) many larger projects have a lot of intra-project dependencies and few third-party ones. At the same time, CI tends to check the entire project, so we're checking most function bodies already, anyway.

I did consider the `go generate` approach, but it isn't currently worth the extra complexity. Maybe it will be in the future, or maybe we'll implement a generic caching layer, or maybe a related project will.
